### PR TITLE
bpfd-operator: Make the CSI deployment default for bpfd-operator

### PR DESCRIPTION
**Pull request by Billy99**
Support for implementing the Container Storage Interface Specification is already in-place. Update the deployment to use it by default and modify the examples to use it as well. Documentation also needs to be updated.

With this change:
- When bpfd is started on the host, CSI will be off. This may be changed in the future, but more work needs to be done to allow in non-K8s.
- When bpfd is started by the bpfd-operator (via daemonset.yaml), it will set the `--csi-support` flag on bpfd, enabling CSI.
   - There is a daemonset-no-csi.yaml if CSI needs to be disabled in a K8 deployment (CSI forces bpfd to K8s v1.26 or later due to K8s bug)
   - `make run-on-kind` now uses CSI.
   - All examples have been updated to use CSI. However, there isn't any logic to flag a Userspace pod that is not using CSI.

Fixes: #776

### CODE REVIEW

**Comment by Billy99** [comment]
@astoycos Ready for review.

/hold
Adding a hold so that it doesn't merge until I remove the first commit (`[DO NOT MERGE] Use updated examples from private repo`)

**Review by astoycos — CHANGES_REQUESTED** [review]
OK @Billy99 This generally looks good just a few things! 

- We don't need to maintain a no-csi deployment, from a K8s POV the user shouldn't even know we're making use of that :) 

- Please remove the maps field from the `bpfProgram` CRD since we no longer populate it anyways (so that the bpfd-agent can be unpriviledged) https://github.com/bpfd-dev/bpfd/blob/main/bpfd-operator/apis/v1alpha1/bpfprogram_types.go#L49.  Eventually maps will be replaced with another CRD but the owner reference for that will be the parent *Program not the node level BpfProgram

- In the bpfd-operator helpers file please remove the get_maps function AND remove all of the commented out code there I don't think we need those anymore.

File: bpfd-operator/config/bpfd-deployment/daemonset-no-csi.yaml [hunk 1]
```diff
@@ -98,28 +88,6 @@ spec:
           mountPath: /etc/bpfd/bpfd.toml
           subPath: bpfd.toml
           readOnly: true
```
[the remark this answers is not in the document]
Comment (Billy99) [reply bpfd-operator/config/bpfd-deployment/daemonset-no-csi.yaml:90]: So my only thought here was with CSI we are forced to use K8s v1.26 or higher. So this allows bpfd to run on older versions of K8s.

File: bpfd-operator/Makefile [hunk 2]
```diff
@@ -324,19 +324,19 @@ uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified
...
-deploy: manifests kustomize ## Deploy bpfd-operator to the K8s cluster specified in ~/.kube/config.
+deploy: manifests kustomize ## Deploy bpfd-operator to the K8s cluster specified in ~/.kube/config with the optional csi driver initialized.
 	cd config/bpfd-operator-deployment && $(KUSTOMIZE) edit set image quay.io/bpfd/bpfd-operator=${BPFD_OPERATOR_IMG}
 	cd config/bpfd-deployment && $(KUSTOMIZE) edit set image quay.io/bpfd/bpfd=${BPFD_IMG} && $(KUSTOMIZE) edit set image quay.io/bpfd/bpfd-agent=${BPFD_AGENT_IMG}
 	$(KUSTOMIZE) build config/default | kubectl apply -f -
 
-## Deploy bpfd with its CSI driver initialized.
-.PHONY: deploy-csi
-deploy-csi: manifests kustomize cm-verifier ## Deploy bpfd-operator to the K8s cluster specified in ~/.kube/config with the optional csi driver initalized.
+## Default deploy target is KIND based without CSI driver initialized.
```
[the remark this answers is not in the document]
Comment (Billy99) [reply bpfd-operator/Makefile:334]: Done

File: bpfd-operator/Containerfile.bpfd-operator [hunk 3]
```diff
@@ -32,7 +32,7 @@ RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o bp
 FROM gcr.io/distroless/static:nonroot
 WORKDIR /
 COPY --from=bpfd-operator-build /usr/src/bpfd/bpfd-operator/config/bpfd-deployment/daemonset.yaml ./config/bpfd-deployment/daemonset.yaml
-COPY --from=bpfd-operator-build /usr/src/bpfd/bpfd-operator/config/bpfd-deployment/daemonset-csi.yaml ./config/bpfd-deployment/daemonset-csi.yaml
+COPY --from=bpfd-operator-build /usr/src/bpfd/bpfd-operator/config/bpfd-deployment/daemonset-no-csi.yaml ./config/bpfd-deployment/daemonset-no-csi.yaml
```
[the remark this answers is not in the document]
Comment (Billy99) [reply bpfd-operator/Containerfile.bpfd-operator:35]: Done

File: bpfd-operator/pkg/helpers/helpers.go [hunk 4]
```diff
@@ -142,42 +138,53 @@ func GetClientOrDie() *bpfdclientset.Clientset {
 // TODO(astoycos) This will completely be removed with the transition to CSI.
 // GetMaps is meant to be used by applications wishing to use BPFD. It takes in a bpf program
 // name and a list of map names, and returns a map corelating map name to map pin path.
-func GetMaps(c *bpfdclientset.Clientset, ProgramName string, mapNames []string) (map[string]string, error) {
+func GetMaps(c *bpfdclientset.Clientset, ProgramName string, mapPinPath string, mapNames []string) (map[string]string, error) {
```
[the remark this answers is not in the document]
Comment (Billy99) [reply bpfd-operator/pkg/helpers/helpers.go:141]: Done

File: bpfd-operator/internal/constants.go [hunk 5]
```diff
@@ -36,7 +36,7 @@ const (
 	BpfdConfigName              = "bpfd-config"
 	BpfdCsiDriverName           = "csi.bpfd.dev"
 	BpfdDaemonManifestPath      = "./config/bpfd-deployment/daemonset.yaml"
-	BpfdDaemonCsiManifestPath   = "./config/bpfd-deployment/daemonset-csi.yaml"
+	BpfdDaemonNoCsiManifestPath = "./config/bpfd-deployment/daemonset-no-csi.yaml"
```
[the remark this answers is not in the document]
Comment (Billy99) [reply bpfd-operator/internal/constants.go:39]: Done

File: bpfd-operator/controllers/bpfd-operator/configmap.go [hunk 6]
```diff
@@ -47,7 +47,8 @@ import (
 type BpfdConfigReconciler struct {
 	ReconcilerCommon
 	BpfdStandardDeployment string
-	BpfdCsiDeployment      string
+	BpfdNoCsiDeployment    string
```
[the remark this answers is not in the document]
Comment (Billy99) [reply bpfd-operator/controllers/bpfd-operator/configmap.go:50]: Done

File: bpfd-operator/config/bpfd-deployment/daemonset-no-csi.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Billy99) [reply bpfd-operator/config/bpfd-deployment/daemonset-no-csi.yaml:90]: Done

**Comment by Billy99** [comment]
@astoycos I think I addressed all your comments and CI is Green. When you are happy with it, I'll remove the first commit.

**Review by astoycos — APPROVED** [review]
LGTM

Thanks @Billy99!!!

**Pull request merged**